### PR TITLE
[move source lang] Allowed generic resources to interact with global storage

### PR DESCRIPTION
**Pull request by tnowacki**
## Motivation

- Refactored Acquires to handling just names of resources, not types
- Acquiring a resources acquires all possible instantiations/instances

## Test Plan

cargo test

## Related PRs

#2228

### CODE REVIEW

File: language/move-lang/src/typing/globals.rs [hunk 1]
```diff
@@ -223,110 +217,94 @@ fn builtin_function(
...
     F: Fn() -> String,
 {
-    check_global_access_(context, loc, msg, global_type, true)
+    check_global_access_(context, loc, msg, global_type)
 }
 
-fn valid_acquires_annot(context: &mut Context, loc: &Loc, global_type: &BaseType) -> bool {
-    let msg = || panic!("ICE should not have recorded errors");
-    check_global_access_(context, loc, msg, global_type, false)
+fn valid_acquires_annot(_context: &mut Context, _loc: Loc, _global_type: &StructName) -> bool {
```
Comment (sblackshear) [inline language/move-lang/src/typing/globals.rs]: Kill?

**Comment by tnowacki** [comment]
>  Are there already tests that attempting to acquire structs (both nominal and structural) fails with the expected message? 

Yup :)

File: language/move-lang/src/typing/globals.rs (hunk 1 above)
Comment (tnowacki) [reply language/move-lang/src/typing/globals.rs]: I wanted to leave the flow in place as there might be some changes to these rules in the future

**Pull request closed without merging**